diff --git a/pkg/build/apk.go b/pkg/build/apk.go
index fbe746b..822e4a3 100644
--- a/pkg/build/apk.go
+++ b/pkg/build/apk.go
@@ -33,6 +33,7 @@ func (bc *Context) initializeApk(ctx context.Context) error {

 	eg.Go(func() error {
 		keyring := sets.List(sets.New(bc.ic.Contents.Keyring...).Insert(bc.o.ExtraKeyFiles...))
+		// fmt.Println("KEYRINGS", keyring)
 		if err := bc.apk.InitKeyring(ctx, keyring, nil); err != nil {
 			return fmt.Errorf("failed to initialize apk keyring: %w", err)
 		}
@@ -65,7 +66,7 @@ func (bc *Context) initializeApk(ctx context.Context) error {
 var repoRE = regexp.MustCompile(`^http[s]?://.+\/alpine\/([^\/]+)\/[^\/]+$`)

 func parseOptionsFromRepositories(repos []string) []string {
-	var versions = make([]string, 0)
+	versions := make([]string, 0)
 	for _, r := range repos {
 		parts := repoRE.FindStringSubmatch(r)
 		if len(parts) < 2 {
